js/views/onboardingModal.js

```diff
@@ -0,0 +1,438 @@
+    return true;
+  },
+
+  accNextKeypress: function(e) {
+    this.triggerOnEnterSpace(e, this.accNext.bind(this));
+  },
+
+  accPrevKeypress: function(e) {
+    this.triggerOnEnterSpace(e, this.accPrev.bind(this));
+  },
+
+  settingsDone: function(e){
+    var self = this,
+        guidCreation;
+
+    guidCreation = this.options.guidCreationPromise;
+
+    if (guidCreation.state() == 'pending') {
+      this.guidStillCreatingModal && guidStillCreatingModal.remove();
+      this.guidStillCreatingModal = new guidStillCreatingModal();
+      this.guidStillCreatingModal.render().open();
+    }
+
+    guidCreation.done(function() {
+      if (self.isRemoved()) return;
+
+      self._settingsDone(e);
+      self.guidStillCreatingModal && self.guidStillCreatingModal.remove();
+    });
+  },
+
+  _settingsDone: function(e){
+    var self = this,
+        server = this.model.get('serverUrl'),
+        profileFormData = new FormData(),
+        settingsFormData = new FormData(),
+        uploadImageFormData = new FormData();
+
+    if(this.$('textarea#aboutInput').val() != ""){
+        self.model.set('short_description', this.$('textarea#aboutInput').val());
+    }
+
+    if(this.$('#storeHandleInput').val() != "" && /^@/.test(this.$('#storeHandleInput').val()) ){
+        self.model.set('handle', this.$('#storeHandleInput').val());
+    }
+
+    if(this.$('#storeNameInput').val() != ""){
+        self.model.set('name', this.$('#storeNameInput').val());
+    }else if (self.model.get('name') == undefined){
+        //otherwise error since the profile api needs the name parameter and as of now it is not set in the userMd.js
+        self.model.set('name', "ob" + Math.random().toString(36).slice(2));
+    }
+
+    var themeId = this.$('input[name=theme-selection]:checked');
+    if(themeId.length > 0){
+      var header = themeId.data('header');
+      var primaryColor = parseInt(themeId.data('primary-color').slice(1), 16);
+      var secondaryColor = parseInt(themeId.data('secondary-color').slice(1), 16);
+      var backgroundColor = parseInt(themeId.data('background-color').slice(1), 16);
+      var textColor = parseInt(themeId.data('text-color').slice(1), 16);
+
+      self.model.set('primary_color', primaryColor);
+      self.model.set('secondary_color', secondaryColor);
+      self.model.set('text_color', textColor);
+      self.model.set('background_color', backgroundColor);
+    }
+
+    $.each(this.model.attributes,
+        function(i,el) {
+            if(i == "country") {
+                profileFormData.append("location",el);
+            }
+            if(i == "name" || i == "handle" || i =="short_description"|| (themeId && (i == "primary_color" || i == "secondary_color" || i == "text_color"|| i =="background_color" ))) {
+                profileFormData.append(i,""+el);
+            } else {
+                settingsFormData.append(i,el);
+            }
+        }
+    );
+
+    var imageURI = this.$('#image-cropper').cropit('export', {
```

Comment:
Until the accordion is ready, the onboarding modal is hidden and the loading modal is visible. That's why the dimensions were coming back as 0. I think moving it to the end of accordion ready is the perfect fix. I've done just that and will push soon.